### PR TITLE
Create skill search

### PR DESCRIPTION
**Pull request by Arturnix**
Added search bar with logic and view to display searched skills on selected hero class.

### CODE REVIEW

File: src/main/java/pl/arturzgodka/controllers/SkillsController.java [hunk 1]
```diff
@@ -24,13 +31,34 @@ public String getHeroesList(Model model) {
...
+        List<HeroSkillDataModel> skillsMatched = getSkillsToDisplay(heroClassName, skillSearchName);
+
+        model.addAttribute("skills", skillsMatched);
+        model.addAttribute("heroClass", heroClassName);
+        model.addAttribute("skillSearched", true); //flag to not display search bar after search was submitted and results are being displayed
+
+        return "skills";
+    }
+
+    private List<HeroSkillDataModel> getSkillsToDisplay(String heroClassName, String skillSearchName) {
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/controllers/SkillsController.java:54]: Should this really be in a controller?

**Review by blazejprzyluski — COMMENTED** [review]
Please include tests for hero search, I'd very much recommend even creating a seperate class just for searching functionality

File: src/main/java/pl/arturzgodka/controllers/SkillsController.java (hunk 1 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/controllers/SkillsController.java]: It is additional method, I moved it out from controller to seperate class just for searching functionality: SearchSkill class.

**Comment by Arturnix** [comment]
Tests included.

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**